generator/graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/generator/internal/types/utils/functionReturnTypes.kt

```diff
@@ -45,7 +45,7 @@ import kotlin.reflect.full.createType
  */
 internal fun getWrappedReturnType(returnType: KType): KType {
     return when {
-        returnType.isSubclassOf(DataFetcherResult::class) -> returnType.getTypeOfFirstArgument()
+        returnType.isSubclassOf(DataFetcherResult::class) -> getWrappedReturnType(returnType.getTypeOfFirstArgument())
```

Comment:
I believe we probably would need to make a distinction between (and add handling for each)

* `DataFetcherResult<T>` - current logic
* `DataFetcherResult<List<T>>` - currently unsupported
* `DataFetcher<List<DataFetcher<T>>>` - currently unsupported

Haven't tested it but following two cases might cover it

```kotlin
  // if return type is DataFetcher<List<T>> check if T has to be unwrapped
  returnType.isSubclassOf(DataFetcherResult::class) && returnType.getTypeOfFirstArgument().isSubclassOf(List::class) -> {
    List::class.createType(listOf(KTypeProjection(KVariance.INVARIANT, getWrappedReturnType(returnType.getTypeOfFirstArgument().getTypeOfFirstArgument()))))
  }
  // keep current case as is
  returnType.isSubclassOf(DataFetcherResult::class) -> returnType.getTypeOfFirstArgument() is
```

Reply:
Actually, I think just doing the `getWrappedReturnType()` as you suggested might do the trick (list types would be handled recursively). Can you check the test failures?

```
GenerateFunctionTest > DataFetcherResult of a CompletableFuture is invalid() FAILED
    org.opentest4j.AssertionFailedError at GenerateFunctionTest.kt:276

FunctionReturnTypesKtTest > getWrappedReturnType of DataFetcherResult() FAILED
    org.opentest4j.AssertionFailedError at FunctionReturnTypesKtTest.kt:66
```

Reply:
Strange. I thought all the tests were passing when I ran locally. There quite a lot of noise with warnings etc, so I might have failed to spot them.

Sorry about this, I will check this out straight away.

Reply:
Provided a fix for the tests. Can you please rerun the workflow?